sklearn/ensemble/tests/test_gradient_boosting.py

```diff
@@ -83,106 +83,115 @@ def check_classification_toy(loss):
 
     # deviance requires ``n_classes >= 2``.
-    assert_raises(ValueError,
-                  lambda X, y: GradientBoostingClassifier(
-                      loss='deviance').fit(X, y),
-                  X, [0, 0, 0, 0])
+    with pytest.raises(ValueError):
+        (lambda X, y: GradientBoostingClassifier(loss='deviance').
+         fit(X, y))(X, [0, 0, 0, 0])
+
+    allowed_presort = ('auto', True, False)
+    err_msg = ("'presort' should be in {}. "
+               "Got 'invalid' instead.".format(allowed_presort))
+    with pytest.raises(ValueError, match=re.escape(err_msg)):
+        GradientBoostingClassifier(presort='invalid').fit(X, y)
 
 
 def test_regressor_parameter_checks():
     # Check input parameter validation for GradientBoostingRegressor
-    assert_raise_message(ValueError, "alpha must be in (0.0, 1.0) but was 1.2",
-                         GradientBoostingRegressor(loss='huber', alpha=1.2)
-                         .fit, X, y)
-    assert_raise_message(ValueError, "alpha must be in (0.0, 1.0) but was 1.2",
-                         GradientBoostingRegressor(loss='quantile', alpha=1.2)
-                         .fit, X, y)
-    assert_raise_message(ValueError, "Invalid value for max_features: "
-                         "'invalid'. Allowed string values are 'auto', 'sqrt'"
-                         " or 'log2'.",
-                         GradientBoostingRegressor(max_features='invalid').fit,
-                         X, y)
-    assert_raise_message(ValueError, "n_iter_no_change should either be None"
-                         " or an integer. 'invalid' was passed",
-                         GradientBoostingRegressor(n_iter_no_change='invalid')
-                         .fit, X, y)
+    with pytest.raises(ValueError, match=r"alpha must be in \(0.0, 1.0\) "
+                                         r"but was 1.2"):
+        GradientBoostingRegressor(loss='huber', alpha=1.2).fit(X, y)
+    with pytest.raises(ValueError, match=r"alpha must be in \(0.0, 1.0\) "
+                                         r"but was 1.2"):
+        GradientBoostingRegressor(loss='quantile', alpha=1.2).fit(X, y)
+    err_msg = ("Invalid value for max_features: "
+               "'invalid'. Allowed string values are 'auto', 'sqrt'"
+               " or 'log2'.")
+    with pytest.raises(ValueError, match=err_msg):
+        GradientBoostingRegressor(max_features='invalid').fit(X, y)
+    err_msg = ("n_iter_no_change should either be None"
+               " or an integer. 'invalid' was passed")
+    with pytest.raises(ValueError, match=err_msg):
+        GradientBoostingRegressor(n_iter_no_change='invalid').fit(X, y)
+    allowed_presort = ('auto', True, False)
+    err_msg = ("'presort' should be in {}. "
+               "Got 'invalid' instead.".format(allowed_presort))
+    with pytest.raises(ValueError, match=re.escape(err_msg)):
+        GradientBoostingRegressor(presort='invalid').fit(X, y)
```

Comment:
@sameshl, I think you should remove this line because **presort** is going to be removed in the future version 0.24. So, don't include the code for the elements which will be depricated soon.